networkx/generators/tests/test_trees.py

```diff
@@ -67,6 +67,86 @@ def source_label(v):
         assert source_label(ten) == "n"
         assert source_label(NIL) == "NIL"
 
+    def test_basic_recursive(self):
```

Comment:
AFAICT this test is identical to the previous one - in that case, you could instead parametrize the test on the prefix tree function - something like:

```python
@pytest.mark.parametrize("prefix_tree_fn", (nx.prefix_tree, nx.prefix_tree_recursive))
def test_prefix_tree(self, prefix_tree_fn):
    ...
    T = prefix_tree_fn(strings)
    ...
```

This has the advantage of:
 1. Making the code significantly shorter, and more importantly
 2. Reducing duplication so that any future modification to the test doesn't have to be done in two separate places.